src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java

```diff
@@ -0,0 +1,449 @@
+     */
+    @Deprecated
+    @Override
+    public IBlockState getStateFromMeta(final int meta)
+    {
+        final EnumFacing enumFacing = EnumFacing.getHorizontal(meta);
+        return this.getDefaultState().withProperty(FACING, enumFacing);
+    }
+
+    /**
+     * @deprecated but we still need this because there is nothing better.
+     */
+    @Override
+    public IBlockState getActualState(final IBlockState state, final IBlockAccess worldIn, final BlockPos pos)
+    {
+        final TileEntity entity = worldIn.getTileEntity(pos);
+
+        if(!(entity instanceof TileEntityRack))
+        {
+            return super.getActualState(state, worldIn, pos);
+        }
+
+        final TileEntityRack rack = (TileEntityRack) entity;
+        if (rack.isEmpty() && (rack.getOtherChest() == null || rack.getOtherChest().isEmpty()))
+        {
+            if(rack.getOtherChest() != null)
+            {
+                if(rack.isMain())
+                {
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, BlockMinecoloniesRack.EnumType.DEFAULTDOUBLE)
+                            .withProperty(FACING, BlockPosUtil.getFacing(rack.getNeighbor(), pos));
+                }
+                else
+                {
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, EnumType.EMPTYAIR);
+                }
+            }
+            else
+            {
+                return state.withProperty(BlockMinecoloniesRack.VARIANT, EnumType.DEFAULT);
+            }
+        }
+        else
+        {
+            if(rack.getOtherChest() != null)
+            {
+                if(rack.isMain())
+                {
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, EnumType.FULLDOUBLE)
+                            .withProperty(FACING, BlockPosUtil.getFacing(rack.getNeighbor(), pos));
+                }
+                else
+                {
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, BlockMinecoloniesRack.EnumType.EMPTYAIR);
+                }
+            }
+            else
+            {
+                return state.withProperty(BlockMinecoloniesRack.VARIANT, EnumType.FULL);
+            }
+        }
+    }
+
+    /**
+     * Check if a certain block should be replaced with a rack.
+     * @param block the block to check.
+     * @return true if so.
+     */
+    public static boolean shouldBlockBeReplacedWithRack(final Block block)
+    {
+        return block == Blocks.CHEST;
+    }
+
+    @Override
+    public int getMetaFromState(@NotNull final IBlockState state)
+    {
+        return state.getValue(FACING).getHorizontalIndex();
+    }
+
+    /**
+     * On the change of a neighbor block.
+     *
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    public void neighborChanged(final IBlockState state, final World worldIn, final BlockPos pos, final Block blockIn)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @Deprecated annotation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)